pumpkin/src/command/commands/cmd_transfer.rs

```diff
@@ -0,0 +1,140 @@
+
+const DESCRIPTION: &str = "Triggers a transfer of a player to another server.";
+
+const ARG_HOSTNAME: &str = "hostname";
+
+const ARG_PLAYERS: &str = "players";
+
+static PORT_CONSUMER: BoundedNumArgumentConsumer<i32> = BoundedNumArgumentConsumer::new()
+    .name("port")
+    .min(1)
+    .max(65535);
+
+struct TransferTargetSelf;
+
+#[async_trait]
+impl CommandExecutor for TransferTargetSelf {
+    async fn execute<'a>(
+        &self,
+        sender: &mut CommandSender<'a>,
+        _server: &crate::server::Server,
+        args: &ConsumedArgs<'a>,
+    ) -> Result<(), CommandError> {
+        let Some(Arg::Simple(hostname)) = args.get(ARG_HOSTNAME) else {
+            return Err(InvalidConsumption(Some(ARG_HOSTNAME.into())));
+        };
+
+        let port = match PORT_CONSUMER.find_arg_default_name(args) {
+            Err(_) => 25565,
+            Ok(Ok(count)) => count,
+            Ok(Err(())) => {
+                sender
+                    .send_message(
+                        TextComponent::text("Port must be between 1 and 65535.")
+                            .color(Color::Named(NamedColor::Red)),
+                    )
+                    .await;
+                return Ok(());
+            }
+        };
+
+        if let CommandSender::Player(player) = sender {
+            let name = &player.gameprofile.name;
+            log::info!("[{name}: Transferring {name} to {hostname}:{port}]");
+            player
+                .client
+                .send_packet(&CTransfer::new(hostname, &VarInt(port)))
+                .await;
+            Ok(())
+        } else {
+            Err(InvalidRequirement)
+        }
+    }
+}
+
+struct TransferTargetPlayer;
+
+#[async_trait]
+impl CommandExecutor for TransferTargetPlayer {
+    async fn execute<'a>(
+        &self,
+        sender: &mut CommandSender<'a>,
+        _server: &crate::server::Server,
+        args: &ConsumedArgs<'a>,
+    ) -> Result<(), CommandError> {
+        let Some(Arg::Simple(hostname)) = args.get(ARG_HOSTNAME) else {
+            return Err(InvalidConsumption(Some(ARG_HOSTNAME.into())));
+        };
+
+        let port = match PORT_CONSUMER.find_arg_default_name(args) {
+            Err(_) => 25565,
+            Ok(Ok(count)) => count,
+            Ok(Err(())) => {
+                sender
+                    .send_message(
+                        TextComponent::text("Port must be between 1 and 65535.")
+                            .color(Color::Named(NamedColor::Red)),
+                    )
+                    .await;
+                return Ok(());
+            }
+        };
+
+        let Some(Arg::Players(players)) = args.get(ARG_PLAYERS) else {
+            return Err(InvalidConsumption(Some(ARG_PLAYERS.into())));
+        };
+
+        let sender_name = match sender {
+            CommandSender::Console => "Console",
+            CommandSender::Rcon(_) => "Rcon",
+            CommandSender::Player(p) => &p.gameprofile.name,
+        };
```

Comment:
I see this is used multiple times in the code base. I think it would make sense to implement `fmt::Display` for CommandSender